src/pages/ProductDetails.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import * as api from '../services/api';
+
+export default class ProductDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    const { id: productId } = props;
+
+    this.state = {
+      title: '',
+      price: '',
+      thumbnail: '',
+      attributes: '',
+      loading: true,
+      id: productId,
```

Comment:
aqui eu acho que poderia ser atribuído o props.id sem o meio campo feito na linha 7. como o valor vai ser atribuído no estado, acho que não precisaria descontruílo e aplicar um alias

```suggestion
      id: props.id,
```